### DISCUSSION TITLE
Include component in classname

### DISCUSSION THREAD

**Discussion by mikematos84** [emotion-js/emotion#2789] [Q&A]
I have been looking around to see if I can find the answer to this but so far have come up short. Comping from styled components, I was used to seeing the component name as part of the className. I was wondering if there was a way to have that same functionality within the classNames generated by emotion CSS. 

I am using @emotion/styled and @emotion/react in my project.

**Comment by srmagura** [comment]
The generated class names will include the component name if you use `@emotion/babel-plugin`. https://github.com/emotion-js/emotion/tree/main/packages/babel-plugin#autolabel

For `@emotion/react` (css prop), we are often able to determine the component name at runtime, so without the Babel plugin's involvement.

You can also pass in `label: 'MyComponent'` to `styled` as if it were a CSS property.
